libminifi/test/rocksdb-tests/RepoTests.cpp

```diff
@@ -515,4 +519,221 @@ TEST_CASE("FlowFileRepository synchronously pushes existing flow files") {
+  REQUIRE(expected_max_repo_size == repository->getMaxRepositorySize());
+  REQUIRE(2 == repository->getRepositoryEntryCount());
+}
+
+TEST_CASE("Test getting noop repository size properties", "[TestGettingRepositorySize]") {
+  TestController testController;
+  auto dir = testController.createTempDirectory();
+
+  auto repository = minifi::core::createRepository("NoOpRepository", "ff");
+
+  repository->initialize(std::make_shared<minifi::Configure>());
+
+  auto flow_file = std::make_shared<minifi::FlowFileRecord>();
+
+  flow_file->addAttribute("key", "testattributevalue");
+
+  repository->flush();
+  repository->stop();
+
+  REQUIRE(repository->getRepositorySize() == 0);
+  REQUIRE(!repository->isFull());
+  REQUIRE(repository->getMaxRepositorySize() == 0);
+  REQUIRE(repository->getRepositoryEntryCount() == 0);
+}
+
+TEST_CASE("Test getting content repository size properties", "[TestGettingRepositorySize]") {
+  LogTestController::getInstance().setDebug<core::ContentRepository>();
+  LogTestController::getInstance().setDebug<core::repository::FileSystemRepository>();
+  LogTestController::getInstance().setDebug<core::repository::VolatileContentRepository>();
+  LogTestController::getInstance().setDebug<core::repository::DatabaseContentRepository>();
+  TestController testController;
+  auto dir = testController.createTempDirectory();
+
+  auto repository = std::make_shared<core::repository::FlowFileRepository>("ff", dir.string(), 0ms, 0, 1ms);
+
+  auto content_repo_dir = testController.createTempDirectory();
+  auto configuration = std::make_shared<minifi::Configure>();
+  configuration->set(minifi::Configure::nifi_dbcontent_repository_directory_default, content_repo_dir.string());
+  std::string content = "content";
+  configuration->set(minifi::Configure::nifi_volatile_repository_options_content_max_bytes, std::to_string(content.size()));
+
+  std::shared_ptr<core::ContentRepository> content_repo;
+  auto expected_is_full = false;
+  uint64_t expected_max_repo_size = 0;
+  SECTION("FileSystemRepository") {
+    content_repo = std::make_shared<core::repository::FileSystemRepository>();
+  }
+
+  SECTION("VolatileContentRepository") {
+    content_repo = std::make_shared<core::repository::VolatileContentRepository>("content");
+    expected_is_full = true;
+    expected_max_repo_size = content.size();
+  }
+
+  SECTION("DatabaseContentRepository") {
+    content_repo = std::make_shared<core::repository::DatabaseContentRepository>();
+  }
+
+  content_repo->initialize(configuration);
+
+  repository->initialize(configuration);
+  repository->loadComponent(content_repo);
+  auto original_content_repo_size = content_repo->getRepositorySize();
+
+  auto flow_file = std::make_shared<minifi::FlowFileRecord>();
+
+  auto content_session = content_repo->createSession();
+  auto claim = content_session->create();
+  auto stream = content_session->write(claim);
+  stream->write(reinterpret_cast<const uint8_t*>(content.data()), content.size());
```

Comment:
```suggestion
  stream->write(gsl::make_span(content).as_span<const std::byte>());
```

Reply:
Updated in d192f3dadee8367c0b10084a228f36ca5bf6d457